module/os/linux/zfs/zvol_os.c

```diff
@@ -486,6 +484,16 @@ zvol_request_impl(zvol_state_t *zv, struct bio *bio, struct request *rq,
 	uint64_t size = io_size(bio, rq);
 	int rw;
 
+	/*
+	 * We don't enable secure erase support for our zvols, so this is
+	 * purely a sanity check.
+	 */
+	if (unlikely(io_is_secure_erase(bio, rq))) {
+		ASSERT(io_is_secure_erase(bio, rq));
```

Comment:
Is is this assert can never fail.  We should keep either the `if` conditional for this check, or the ASSERT.

Reply:
> As is this assert can never fail

The assert will be ignored unless the user does `./configure --enable-debug`.

Reply:
@amotin yea that's a better idea.  I'll do that in my next push